rdflib/plugins/memory.py

```diff
@@ -49,6 +52,15 @@ def add(self, triple, context, quoted=False):
         # exits.
         subject, predicate, object = triple
         spo = self.__spo
+
+        try:
+            # Triple already in store
+            if(spo[subject][predicate][object]==1):
```

Comment:
Like the fast key tests below, you can replace this without an if statement like so:
```python
try:
    # Triple already in store
    _ = spo[subject][predicate][object]
except:
    # New triple being added
    self.__size += 1
```

Reply:
Don't need to check if the value `== 1`, because when the triple is removed, the whole key+value is removed from the corresponding dict.

Reply:
Yes, that makes sense. I'll change it.